### ISSUE TITLE
not working on thunderbird verstion 86.0b1

### ISSUE THREAD

**Issue by bonbonbillo** [protz/Manually-Sort-Folders#140]
not working on thunderbird verstion 86.0b1

**Comment by globewalldesk** [comment]
Can confirm. Just installed, restarted, and "Manually Sort Folders" is not appearing under hamburger > Tools.

**Comment by MorningSleeper** [comment]
+1, would really like to see support for this if possible.

**Comment by Laulo-77** [comment]
Hello,

+1, not working on thunderbird version 86.0b1

Would it be possible to have a compatible version?

Thank you.

**Comment by Qaezar** [comment]
Hello, +1

Can you please update your addon ? ^^
Thanks ! :)

**Comment by blono** [comment]
When I looked at the console, I got an error in the following.
Array.from(...) .find(...) .xulBrowser.contentWindow is null
Probably it is affected by the following.
https://developer.thunderbird.net/add-ons/mailextensions/experiments

> This hack only works because Thunderbird is internally not (yet) using multiple processes. Again, it is highly recommended to design your APIs in a way that these interactions are not necessary as it is likely that this technique will stop working in future versions of Thunderbird.

**Comment by itagagaki** [comment]
@blono I'm sure your reports will be useful.

Jonathan, the author of this add-on, has not been around for a while.
I've tried to contact him by email, but he hasn't responded.

I'm thinking of helping with modifying this add-on to be TB86 compatible, 
but since TB86 is still in beta and it works fine with TB78, I'm not sure that's necessary yet.

Is there any reason why you want to use TB86 beta instead of TB78?

**Comment by schneeschmelze** [comment]
Am 12.02.21 um 10:15 Uhr schrieb Itagaki Fumihiko:
> but since TB86 is still in beta and it works fine with TB78, I'm not
> sure that's necessary yet.


I think we should be aware that TB86 will be released anytime soon...

**Comment by wsmwk** [comment]
> I think we should be aware that TB86 will be released anytime soon...

@schneeschmelze Do you mean NOT released anytime soon? 

Which is accurate - the next major version to be released will be 89 in a few months.
However, this add-on is quite popular, so it can be a barrier to many users' ability to run beta.

**Comment by schneeschmelze** [comment]
Am 12.02.21 um 10:35 Uhr schrieb Wayne Mery:
> the next major version to be released will be 89 in a few months.
> However, this add-on is quite popular, so it can be a barrier to many
> users' ability to run beta.


+1. Thanks for clarifying.

**Comment by wsmwk** [comment]
correction, the next esr is 91 (not 89), in four months

**Comment by blono** [comment]
https://github.com/thundernest/addon-developer-support/issues/56#issuecomment-772071322
This will be a good reference for the new implementation.

>   getMessenger(context) {
>     let localstorage = context.apiCan.findAPIPath("storage");
>     localstorage.local.get = (...args) =>
>       localstorage.local.callMethodInParentProcess("get", args);
>     localstorage.local.set = (...args) =>
>       localstorage.local.callMethodInParentProcess("set", args);
>     localstorage.local.remove = (...args) =>
>       localstorage.local.callMethodInParentProcess("remove", args);
>     localstorage.local.clear = (...args) =>
>       localstorage.local.callMethodInParentProcess("clear", args);
>     
>     return { 
>       "storage": localstorage,
>       "i18n": context.apiCan.findAPIPath("i18n"),      
>     };
>   }

startupJS.WL.messenger = Array.from(self.extension.views).find(
              view => view.viewType === "background").xulBrowser.contentWindow
              .wrappedJSObject.browser;
↓
WL.messenger = self.getMessenger(self.context);

This new implementation was researched using the following search results.
https://github.com/search?q=thunderbird+%2286.0b1%22&type=issues

**Comment by blono** [comment]
https://github.com/protz/Manually-Sort-Folders/pull/141
I was able to confirm that this PR works with Thunderbird 86.0b2.

**Comment by Gluups** [comment]
> 
> 
> Hello,
> 
> +1, not working on thunderbird version 86.0b1
> 
> Would it be possible to have a compatible version?
> 
> Thank you.
 
Hello,
Not sure this has been answered. It would deserve a more efficient forum software to avoid landing in a specific question about a more recent development, but this is another topic.
I just tested : "Manually Sort Folders" 2.0.5 at least works with versions 68 and 78 of Thunderbird, which are current released versions.
I made these tests on Windows 10.

I presume we can anticipate next question : where is it possible to download version 2.0.5 of "Manually Sort Folders" from ?
I am afraid I am not very up-to-date about this.
